types/plugin.d.ts

```diff
@@ -7,12 +7,11 @@ import { RawServerBase, RawServerDefault, RawRequestDefaultExpression, RawReplyD
  *
  * Fastify allows the user to extend its functionalities with plugins. A plugin can be a set of routes, a server decorator or whatever. To activate plugins, use the `fastify.register()` method.
  */
-export interface FastifyPlugin<Options extends FastifyPluginOptions = {}> {
-  (
-    instance: FastifyInstance<RawServerBase, RawRequestDefaultExpression<RawServerBase>, RawReplyDefaultExpression<RawServerBase>>,
-    opts: Options,
-    next: (err?: FastifyError) => void
-  ): void;
+export type FastifyPlugin<Options extends FastifyPluginOptions = {}, FI extends FastifyInstance = FastifyInstance> =
+{
+  (instance: FI, opts: Options, next: (err?: FastifyError) => void): void;
```

Comment:
I think the type for the `err` parameter should be `Error`.

Reply:
That's unchanged from how it was before.
I can change it here but I guess a separate PR would be better.